x-pack/plugins/infra/server/lib/alerting/common/utils.ts

```diff
@@ -230,10 +230,14 @@ export const flattenAdditionalContext = (
   if (additionalContext) {
     Object.keys(additionalContext).forEach((context: string) => {
       if (additionalContext[context]) {
-        flattenedContext = {
-          ...flattenedContext,
-          ...flattenObject(additionalContext[context], [context + '.']),
-        };
+        if (context !== 'tags') {
```

Comment:
As we discussed, I am wondering if we can improve this condition to be more generic, but I don't know how to do so, @simianhacker do you have a suggestion for this?

If we remove this condition, it will generate the following error:
```
{"type":"mapper_parsing_exception",
"reason":"failed to parse field [tags] of type [keyword] in document with id 
'80ff5ec3-9432-45a4-81bf-f2beb5c1e7ec'. Preview of field's value: '{0=Test}'",
"caused_by":{"type":"illegal_state_exception",
"reason":"Can't get text on a START_OBJECT at 1:1418"}
```

Reply:
I don't think we should be flattening array of primitives (like `keywords` or `numbers`). The same thing is happening to the `host.mac` field which ends up like `host.mac[0]` for the first item in the array. If I tried to search for `host.mac: 00-00-00-00-00` it would not match `{ "host.mac[0]": "00-00-00-00-00" }` because the field names are not the same. If it was stored as `{ "host.mac": ['"00-00-00-00-00"] }` then it would match.

I wonder if we shouldn't add an additional argument to the `flattenObject` function that `ignoreArrays`. Right now it has an argument about how to configure the array key but no way to stop it form recursing through the array.

Reply:
Ah, interesting. I was able to index `host.mac[0]` in AAD without any errors, so it seemed it would work. But for `tags`, I was getting the exception. I will refactor the `flattenObject` function to make sure the fields match with ecs fields.

Reply:
@maryam-saeidi I have changed the way to flatten the context. Now it should not flatten the array with `[x]` in keys.